Create a new section like the timeline, but with all the podcasts to which we are subscribed

### Clear and concise description of the problem

It is still difficult to access recently released episodes. Although the filter option has been added, which works great, waiting for the last selected order option to be saved, it is difficult to see the latest episodes published in the rest of the podcasts that are not favorites.

### Suggested solution

Create a new section where the podcast episodes appear in order of publication. The last podcasts would be the first. Something similar to the timeline section but with all the podcasts.



### Alternative

I miss that the timeline option does not appear more days, I understand that it has been limited so that the page does not become heavy, in this case it could generate several podcast numbers in the section and add, for example, a week with the latest published episodes, ordered by date - time of publication (as in timeline) and separated by days up to the last week (as in timeline)

### Additional context

![image](https://user-images.githubusercontent.com/27757712/236693908-a47221e6-dcf5-42b9-8d78-7f9239282b7e.png)


### Validations

- [X] Follow our [Code of Conduct](https://github.com/SamTV12345/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/SamTV12345/.github/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

With the last update of the dev version it does not work

```
2023-05-10T18:41:30 [INFO] - Settings already present
2023-05-10T18:41:30 [INFO] - Starting with the following environment variables:
2023-05-10T18:41:30 [INFO] - starting 4 workers
2023-05-10T18:41:30 [INFO] - Actix runtime found; starting in Actix runtime
2023-05-10T18:41:40 [ERROR] - stream error: Request parse error: Invalid Header provided
2023-05-10T18:41:40 [ERROR] - stream error: Request parse error: Invalid Header provided
2023-05-10T18:41:40 [ERROR] - stream error: Request parse error: Invalid Header provided
2023-05-10T18:41:40 [ERROR] - stream error: Request parse error: Invalid Header provided
2023-05-10T18:41:41 [ERROR] - stream error: Request parse error: Invalid Header provided
2023-05-10T18:41:41 [ERROR] - stream error: Request parse error: Invalid Header provided
2023-05-10T18:41:41 [ERROR] - stream error: Request parse error: Invalid Header provided
```

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,8 +1,0 @@
-[target.x86_64-unknown-linux-gnu]
-rustflags = ["-C", "link-arg=-fuse-ld=lld"]
-
-[env]
-RUST_TEST_THREADS = { value = "300", force = true }
-
-[target.aarch64-unknown-linux-gnu]
-linker = "aarch64-linux-gnu-gcc"

--- a/migrations/2023-05-20-175610_only_favored_podcasts_filter/down.sql
+++ b/migrations/2023-05-20-175610_only_favored_podcasts_filter/down.sql
@@ -1,0 +1,2 @@
+-- This file should undo anything in `up.sql`
+ALTER TABLE filters DROP COLUMN only_favored;

--- a/migrations/2023-05-20-175610_only_favored_podcasts_filter/up.sql
+++ b/migrations/2023-05-20-175610_only_favored_podcasts_filter/up.sql
@@ -1,0 +1,2 @@
+-- Your SQL goes here
+ALTER TABLE filters ADD COLUMN only_favored BOOLEAN NOT NULL DEFAULT TRUE;

--- a/src/controllers/podcast_controller.rs
+++ b/src/controllers/podcast_controller.rs
@@ -68,11 +68,22 @@ pub async fn search_podcasts(query: web::Query<PodcastSearchModel>, conn:Data<Db
     let query = query.into_inner();
     let order = query.order.unwrap_or(OrderCriteria::ASC);
     let latest_pub = query.order_option.unwrap_or(OrderOption::Title);
+    let only_favored;
+    let opt_filter = Filter::get_filter_by_username(requester.clone().unwrap().username.clone(),
+                                                    &mut *conn.get().unwrap()).unwrap();
+    match opt_filter {
+        Some(filter)=>{
+            only_favored = filter.only_favored;
+        },
+        None=>{
+            only_favored = true
+        }
+    }
 
     let username = requester.unwrap().username.clone();
     let filter = Filter::new(username.clone(), query.title.clone(), order.clone().to_bool(),Some
                 (latest_pub.clone()
-                .to_string()));
+                .to_string()),only_favored);
     Filter::save_filter(filter, &mut *conn.get().unwrap()).expect("Error saving filter");
 
     match query.favored_only {

--- a/src/controllers/podcast_episode_controller.rs
+++ b/src/controllers/podcast_episode_controller.rs
@@ -56,6 +56,7 @@ pub struct TimeLinePodcastEpisode {
 #[serde(rename_all = "camelCase")]
 pub struct TimelineQueryParams {
     pub favored_only: bool,
+    pub last_timestamp: Option<String>
 }
 
 #[get("/podcasts/timeline")]
@@ -67,7 +68,7 @@ Responder {
 
 
     let res = DB::get_timeline(requester.unwrap().username.clone(),&mut conn.get().unwrap(),
-                               favored_only.favored_only);
+                               favored_only.into_inner());
 
     let mapped_timeline = res.iter().map(|podcast_episode| {
         let (podcast_episode, podcast) = podcast_episode;

--- a/src/controllers/podcast_episode_controller.rs
+++ b/src/controllers/podcast_episode_controller.rs
@@ -52,15 +52,22 @@ pub struct TimeLinePodcastEpisode {
     podcast: Podcast,
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TimelineQueryParams {
+    pub favored_only: bool,
+}
+
 #[get("/podcasts/timeline")]
 pub async fn get_timeline(conn: Data<DbPool>,  requester: Option<web::ReqData<User>>, mapping_service:
-Data<Mutex<MappingService>>) ->
+Data<Mutex<MappingService>>, favored_only: Query<TimelineQueryParams>) ->
                                                                                              impl
 Responder {
     let mapping_service = mapping_service.lock().ignore_poison();
 
 
-    let res = DB::get_timeline(requester.unwrap().username.clone(),&mut conn.get().unwrap());
+    let res = DB::get_timeline(requester.unwrap().username.clone(),&mut conn.get().unwrap(),
+                               favored_only.favored_only);
 
     let mapped_timeline = res.iter().map(|podcast_episode| {
         let (podcast_episode, podcast) = podcast_episode;

--- a/src/db.rs
+++ b/src/db.rs
@@ -19,6 +19,7 @@ use std::time::SystemTime;
 use diesel::sql_types::{Text, Timestamp};
 use crate::models::episode::{Episode, EpisodeAction};
 use crate::models::favorites::Favorite;
+use crate::models::filter::Filter;
 use crate::models::order_criteria::{OrderCriteria, OrderOption};
 use crate::utils::do_retry::do_retry;
 use crate::utils::time::opt_or_empty_string;
@@ -782,18 +783,35 @@ impl DB {
     }
 
 
-    pub fn get_timeline(username_to_search: String, conn: &mut SqliteConnection) -> Vec<
+    pub fn get_timeline(username_to_search: String, conn: &mut SqliteConnection, favored_only: bool)
+        -> Vec<
         (PodcastEpisode, Podcast)> {
-        let podcast_timeline = sql_query("SELECT * FROM podcast_episodes,podcasts, \
-        favorites \
-        WHERE podcasts.id = podcast_episodes.podcast_id AND podcasts.id = favorites.podcast_id \
-        AND favorites.username=? AND favored=1 ORDER BY podcast_episodes.date_of_recording DESC \
-        LIMIT 20");
+        Filter::save_decision_for_timeline(username_to_search.clone(),conn,favored_only);
 
-        let res = podcast_timeline.bind::<Text, _>(&username_to_search);
+        match favored_only {
+            true=>{
+                let podcast_timeline = sql_query("SELECT * FROM podcast_episodes,podcasts, \
+                favorites \
+                WHERE podcasts.id = podcast_episodes.podcast_id AND podcasts.id = favorites.podcast_id \
+                AND favorites.username=? AND favored=1 ORDER BY podcast_episodes.date_of_recording DESC \
+                LIMIT 20");
+
+                let res = podcast_timeline.bind::<Text, _>(&username_to_search);
 
 
-        res.load::<(PodcastEpisode, Podcast)>(conn).expect("Error loading podcast episode by id")
+                res.load::<(PodcastEpisode, Podcast)>(conn).expect("Error loading podcast episode by id")
+            }
+            false=>{
+                let podcast_timeline = sql_query("SELECT * FROM podcast_episodes,podcasts \
+                WHERE podcasts.id = podcast_episodes.podcast_id ORDER BY podcast_episodes.date_of_recording DESC \
+                LIMIT 20");
+
+                podcast_timeline.load::<(PodcastEpisode, Podcast)>(conn).expect("Error loading \
+                podcast episode by id")
+            }
+        }
+
+
     }
 
     pub fn get_watch_logs_by_username(username_to_search: String, conn: &mut SqliteConnection,

--- a/src/db.rs
+++ b/src/db.rs
@@ -17,10 +17,12 @@ use std::io::Error;
 use std::sync::MutexGuard;
 use std::time::SystemTime;
 use diesel::sql_types::{Text, Timestamp};
+use crate::controllers::podcast_episode_controller::TimelineQueryParams;
 use crate::models::episode::{Episode, EpisodeAction};
 use crate::models::favorites::Favorite;
 use crate::models::filter::Filter;
 use crate::models::order_criteria::{OrderCriteria, OrderOption};
+use crate::schema::podcast_episodes::dsl::podcast_episodes;
 use crate::utils::do_retry::do_retry;
 use crate::utils::time::opt_or_empty_string;
 
@@ -783,31 +785,55 @@ impl DB {
     }
 
 
-    pub fn get_timeline(username_to_search: String, conn: &mut SqliteConnection, favored_only: bool)
-        -> Vec<
-        (PodcastEpisode, Podcast)> {
-        Filter::save_decision_for_timeline(username_to_search.clone(),conn,favored_only);
+    pub fn get_timeline(username_to_search: String, conn: &mut SqliteConnection, favored_only: TimelineQueryParams)
+        -> Vec<(PodcastEpisode, Podcast)> {
+        use crate::schema::podcast_episodes::dsl::*;
+        use crate::schema::podcasts::dsl::*;
+        use crate::schema::podcasts::id as pid;
+        use crate::schema::podcast_episodes::id as episode_id;
+        use crate::schema::favorites::dsl::*;
+        use crate::schema::favorites::username as f_username;
+        use crate::schema::favorites::podcast_id as f_podcast_id;
+        use crate::schema::podcast_episodes::podcast_id as e_podcast_id;
 
-        match favored_only {
+        Filter::save_decision_for_timeline(username_to_search.clone(),conn,favored_only.favored_only);
+
+        let mut query = podcast_episodes.inner_join(podcasts.on(e_podcast_id.eq(pid)))
+            .left_join(favorites.on(f_username.eq(username_to_search.clone()).and(f_podcast_id.eq(pid))))
+            .order(date_of_recording.desc())
+            .limit(20)
+            .into_boxed();
+
+        match favored_only.favored_only {
             true=>{
-                let podcast_timeline = sql_query("SELECT * FROM podcast_episodes,podcasts, \
-                favorites \
-                WHERE podcasts.id = podcast_episodes.podcast_id AND podcasts.id = favorites.podcast_id \
-                AND favorites.username=? AND favored=1 ORDER BY podcast_episodes.date_of_recording DESC \
-                LIMIT 20");
+                match favored_only.last_timestamp {
+                    Some(last_id) => {
+                        query = query.filter(date_of_recording.lt(last_id));
+                    }
+                    None => {}
+                }
 
-                let res = podcast_timeline.bind::<Text, _>(&username_to_search);
+                query = query.filter(f_username.eq(username_to_search.clone()));
 
-
-                res.load::<(PodcastEpisode, Podcast)>(conn).expect("Error loading podcast episode by id")
+                query.load::<(PodcastEpisode, Podcast, Option<Favorite>)>(conn).expect("Error loading podcast \
+                episode by id").iter().map(|(episode, podcast, _)| {
+                    (episode.clone(), podcast.clone())
+                }).collect()
             }
             false=>{
-                let podcast_timeline = sql_query("SELECT * FROM podcast_episodes,podcasts \
-                WHERE podcasts.id = podcast_episodes.podcast_id ORDER BY podcast_episodes.date_of_recording DESC \
-                LIMIT 20");
+                match favored_only.last_timestamp {
+                    Some(last_id) => {
+                        query = query.filter(date_of_recording.lt(last_id));
+                    }
+                    None => {}
+                }
 
-                podcast_timeline.load::<(PodcastEpisode, Podcast)>(conn).expect("Error loading \
-                podcast episode by id")
+                query.load::<(PodcastEpisode, Podcast, Option<Favorite>)>(conn).expect
+                ("Error \
+                loading \
+                podcast episode by id").iter().map(|(episode, podcast, _)| {
+                    (episode.clone(), podcast.clone())
+                }).collect()
             }
         }
 

--- a/src/db.rs
+++ b/src/db.rs
@@ -22,7 +22,7 @@ use crate::models::episode::{Episode, EpisodeAction};
 use crate::models::favorites::Favorite;
 use crate::models::filter::Filter;
 use crate::models::order_criteria::{OrderCriteria, OrderOption};
-use crate::schema::podcast_episodes::dsl::podcast_episodes;
+
 use crate::utils::do_retry::do_retry;
 use crate::utils::time::opt_or_empty_string;
 
@@ -790,7 +790,7 @@ impl DB {
         use crate::schema::podcast_episodes::dsl::*;
         use crate::schema::podcasts::dsl::*;
         use crate::schema::podcasts::id as pid;
-        use crate::schema::podcast_episodes::id as episode_id;
+        
         use crate::schema::favorites::dsl::*;
         use crate::schema::favorites::username as f_username;
         use crate::schema::favorites::podcast_id as f_podcast_id;

--- a/src/models/filter.rs
+++ b/src/models/filter.rs
@@ -6,21 +6,25 @@ use diesel::AsChangeset;
 use diesel::Queryable;
 
 #[derive(Debug, Clone, Serialize, Deserialize, Insertable, AsChangeset, Queryable)]
+#[serde(rename_all = "camelCase")]
 pub struct Filter{
-    username: String,
-    title: Option<String>,
-    ascending: bool,
-    filter: Option<String>
+    pub username: String,
+    pub title: Option<String>,
+    pub ascending: bool,
+    pub filter: Option<String>,
+    pub only_favored: bool
 }
 
 
 impl Filter{
-    pub fn new(username: String, title: Option<String>, ascending: bool, filter: Option<String>) -> Self{
+    pub fn new(username: String, title: Option<String>, ascending: bool, filter: Option<String>,
+               only_favored: bool) -> Self{
         Filter{
             username,
             title,
             ascending,
-            filter
+            filter,
+            only_favored
         }
     }
 
@@ -47,5 +51,16 @@ impl Filter{
         let opt_filter = filters.filter(username.eq(username1)).first::<Filter>(conn)
             .optional().expect("Error connecting to database"); // delete all filters
         Ok(opt_filter)
+    }
+
+    pub fn save_decision_for_timeline(username_to_search: String, conn: &mut SqliteConnection,
+                                      only_favored_to_insert:
+    bool){
+        use crate::schema::filters::only_favored;
+        use crate::schema::filters::dsl::*;
+        diesel::update(filters.filter(username.eq(username_to_search)))
+            .set(only_favored.eq(only_favored_to_insert))
+            .execute(conn)
+            .expect("Error connecting to database");
     }
 }

--- a/src/schema.rs
+++ b/src/schema.rs
@@ -40,6 +40,7 @@ diesel::table! {
         title -> Nullable<Text>,
         ascending -> Bool,
         filter -> Nullable<Text>,
+        only_favored -> Bool,
     }
 }
 

--- a/ui/src/components/PodcastEpisodeTimeLine.tsx
+++ b/ui/src/components/PodcastEpisodeTimeLine.tsx
@@ -8,13 +8,17 @@ import {FC} from "react";
 import {selectPodcastImage} from "../pages/Homepage";
 import {TimeLineModel} from "../models/TimeLineModel";
 import {useAppDispatch} from "../store/hooks";
+import {Waypoint} from "react-waypoint";
+import {addPodcastEpisodes, addTimelineEpisodes} from "../store/CommonSlice";
 
 
 type PodcastEpisodeTimeLineProps = {
-    podcastEpisode: TimeLineModel
+    podcastEpisode: TimeLineModel,
+    index: number,
+    timelineLength: number
 }
 
-export const PodcastEpisodeTimeLine:FC<PodcastEpisodeTimeLineProps> = ({podcastEpisode}) => {
+export const PodcastEpisodeTimeLine:FC<PodcastEpisodeTimeLineProps> = ({podcastEpisode, index, timelineLength}) => {
     const dispatch = useAppDispatch()
 
     return <div key={podcastEpisode.podcast_episode.episode_id+"dv"}

--- a/ui/src/language/json/de.json
+++ b/ui/src/language/json/de.json
@@ -126,5 +126,6 @@
   "dash-separated": "Bindestrich-getrennt",
   "dash-separated-with-space": "Bindestrich-getrennt mit Abstand",
   "remove": "Entfernen",
-  "standard-podcast-format": "Standard-Podcastformat"
+  "standard-podcast-format": "Standard-Podcastformat",
+  "onlyFavored": "Nur favorisierte Podcasts"
 }

--- a/ui/src/language/json/en.json
+++ b/ui/src/language/json/en.json
@@ -115,6 +115,7 @@
   "sort-by-published-date": "Sort by published date",
   "sort-by-title": "Sort by title",
   "number-of-podcasts-to-download": "Number of initial podcasts to download",
+  "onlyFavored": "Only favored podcasts",
   "podcast-naming": "Podcast naming",
   "rename-podcasts": "Rename podcasts",
   "replace-illegal-characters": "Replace illegal characters",

--- a/ui/src/language/json/fr.json
+++ b/ui/src/language/json/fr.json
@@ -124,5 +124,6 @@
   "dash-separated": "Séparé par des tirets",
   "dash-separated-with-space": "Séparé par des tirets avec espacement",
   "remove": "Supprimer",
-  "standard-podcast-format": "Format standard du podcast"
+  "standard-podcast-format": "Format standard du podcast",
+  "onlyFavored": "Seulement favoris podcasts"
 }

--- a/ui/src/models/Filter.ts
+++ b/ui/src/models/Filter.ts
@@ -2,5 +2,6 @@ export interface Filter {
     username: string
     title: string
     ascending: boolean
-    filter?: string
+    filter?: string,
+    onlyFavored: boolean
 }

--- a/ui/src/pages/Podcasts.tsx
+++ b/ui/src/pages/Podcasts.tsx
@@ -1,8 +1,8 @@
-import {FC, useEffect, useState} from "react";
+import {FC, useEffect} from "react";
 import axios, {AxiosResponse} from "axios";
-import {apiURL} from "../utils/Utilities";
+import {apiURL, getFiltersDefault} from "../utils/Utilities";
 import {useAppDispatch, useAppSelector} from "../store/hooks";
-import {Podcast, setPodcasts} from "../store/CommonSlice";
+import {Podcast, setFilters, setPodcasts} from "../store/CommonSlice";
 import {Card} from "../components/Card";
 import {AddPodcast} from "../components/AddPodcast";
 import {setModalOpen} from "../store/ModalSlice";
@@ -23,11 +23,8 @@ export const Podcasts:FC<PodcastsProps> = ({onlyFavorites})=>{
     const podcasts = useAppSelector(state=>state.common.podcasts)
     const dispatch = useAppDispatch()
     let location = useLocation();
-    const [searchText, setSearchText] = useState<string>('')
-    const [orderOfPodcasts, setOrderOfPodcasts] = useState<Order>(Order.ASC)
-    const [latestPub, setLatestPub] = useState<OrderCriteria>(OrderCriteria.TITLE)
     const {t} = useTranslation()
-
+    const filters = useAppSelector(state=>state.common.filters)
     const refreshAllPodcasts = ()=>{
         axios.post(apiURL+"/podcast/all")
     }
@@ -35,9 +32,9 @@ export const Podcasts:FC<PodcastsProps> = ({onlyFavorites})=>{
     const performFilter =()=>{
         axios.get(apiURL + "/podcasts/search", {
             params: {
-                title: searchText,
-                order: orderOfPodcasts,
-                orderOption: latestPub,
+                title: filters?.title,
+                order: filters?.ascending?Order.ASC:Order.DESC,
+                orderOption: filters?.filter=="PUBLISHEDDATE"?"PUBLISHEDDATE":"TITLE",
                 favoredOnly: !!onlyFavorites
             }
         })
@@ -48,19 +45,22 @@ export const Podcasts:FC<PodcastsProps> = ({onlyFavorites})=>{
 
     useDebounce(()=> {
         performFilter();
-    },500, [searchText, orderOfPodcasts, latestPub])
+    },500, [filters])
 
     useEffect(()=>{
-        axios.get(apiURL+"/podcasts/filter").then((c:AxiosResponse<Filter>)=>{
+        axios.get(apiURL+"/podcasts/filter")
+            .then((c:AxiosResponse<Filter>)=>{
             if(c.data === null){
-                setLatestPub(OrderCriteria.TITLE)
-                setOrderOfPodcasts(Order.ASC)
-                performFilter()
+                dispatch(setFilters(getFiltersDefault()))
             }
             else{
-                setLatestPub(c.data.filter === "PublishedDate"?OrderCriteria.PUBLISHEDDATE:OrderCriteria.TITLE)
-                setOrderOfPodcasts(c.data.ascending?Order.ASC:Order.DESC)
-                c.data.title&&setSearchText(c.data.title)
+                dispatch(setFilters({
+                    ascending: c.data.ascending,
+                    filter: c.data.filter,
+                    title: c.data.title,
+                    username: c.data.username,
+                    onlyFavored: c.data.onlyFavored
+                }))
             }
         })
     },[location])
@@ -70,7 +70,7 @@ export const Podcasts:FC<PodcastsProps> = ({onlyFavorites})=>{
         <AddPodcast/>
         <div className="flex flex-col md:flex-row gap-3">
                 <span className="relative  w-full md:w-1/3">
-                    <input type="text" value={searchText}  onChange={v => setSearchText(v.target.value)}
+                    <input type="text" value={filters?.title} onChange={v => dispatch(setFilters({...filters as Filter,title: v.target.value}))}
                            className="border-gray-400 w-full pl-10 pt-1 pb-1 border-2 rounded-2xl"/>
                     <span className="absolute left-2 top-1.5 scale-90">
                         <MaginifyingGlassIcon/>
@@ -78,13 +78,13 @@ export const Podcasts:FC<PodcastsProps> = ({onlyFavorites})=>{
                 </span>
             <select  className="border text-sm rounded-lg block p-2.5 bg-gray-700 border-gray-600 placeholder-gray-400 text-white focus:ring-blue-500 focus:border-blue-500"
                      onChange={(v)=> {
-                       setLatestPub(v.target.value as OrderCriteria)
-                     }} value={latestPub}>
+                         dispatch(setFilters({...filters as Filter,filter: v.target.value as OrderCriteria}))
+                     }} value={filters?.ascending?Order.ASC:Order.DESC}>
                 <option value={OrderCriteria.PUBLISHEDDATE}>{t('sort-by-published-date')}</option>
                 <option value={OrderCriteria.TITLE}>{t('sort-by-title')}</option>
             </select>
             <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth="1.5"
-                 stroke="currentColor" className={`${orderOfPodcasts==Order.DESC?'rotate-180':''} w-6 h-6`} onClick={()=>{setOrderOfPodcasts(orderOfPodcasts==Order.DESC?Order.ASC:Order.DESC)}}>
+                 stroke="currentColor" className={`${filters?.ascending?'rotate-180':''} w-6 h-6`} onClick={()=>{dispatch(setFilters({...filters as Filter,ascending: !filters?.ascending}))}}>
                 <path strokeLinecap="round" strokeLinejoin="round" d="M4.5 10.5L12 3m0 0l7.5 7.5M12 3v18"/>
             </svg>
                 <div className="flex-1"></div>

--- a/ui/src/pages/Timeline.tsx
+++ b/ui/src/pages/Timeline.tsx
@@ -2,10 +2,13 @@ import {useTranslation} from "react-i18next";
 import {useAppDispatch, useAppSelector} from "../store/hooks";
 import {useEffect, useMemo} from "react";
 import axios, {AxiosResponse} from "axios";
-import {apiURL, formatTime} from "../utils/Utilities";
-import {PodcastEpisode, setTimeLineEpisodes} from "../store/CommonSlice";
+import {apiURL, formatTime, getFiltersDefault} from "../utils/Utilities";
+import {setFilters, setTimeLineEpisodes} from "../store/CommonSlice";
 import {PodcastEpisodeTimeLine} from "../components/PodcastEpisodeTimeLine";
 import {TimeLineModel} from "../models/TimeLineModel";
+import {Switcher} from "../components/Switcher";
+import {Filter} from "../models/Filter";
+import {Loading} from "../components/Loading";
 
 const convertToTimeLineEpisodes = (podcastEpisodes: TimeLineModel[]) => {
     return podcastEpisodes.reduce((groups, game) => {
@@ -26,19 +29,46 @@ export const Timeline = ()=>{
     const dispatch = useAppDispatch()
     const timeLineEpisodes = useAppSelector(state=>state.common.timeLineEpisodes)
     const mappedEpisodes = useMemo(()=>convertToTimeLineEpisodes(timeLineEpisodes),[timeLineEpisodes])
+    const filter = useAppSelector(state=>state.common.filters)
 
     useEffect(()=>{
-        axios.get(apiURL+"/podcasts/timeline")
-            .then((c:AxiosResponse<TimeLineModel[]>)=>{
-                dispatch(setTimeLineEpisodes(c.data))
-            })
-        },
-        [])
+        !filter&&axios.get(apiURL+"/podcasts/filter")
+            .then((filterAxiosResponse:AxiosResponse<Filter>)=>{
+                    filterAxiosResponse.data == null && dispatch(setFilters(getFiltersDefault()))
 
-    console.log(convertToTimeLineEpisodes(timeLineEpisodes))
+                dispatch(setFilters(filterAxiosResponse.data))
+            })
+    }, [])
+
+    useEffect(()=>{
+        if (filter){
+            axios.get(apiURL+"/podcasts/timeline",{
+                params:{
+                    favoredOnly: filter.onlyFavored
+                }
+            })
+                .then((c:AxiosResponse<TimeLineModel[]>)=>{
+                    dispatch(setTimeLineEpisodes(c.data))
+                })
+        }
+    },[filter])
+
+    if (filter == null){
+        return <Loading/>
+    }
 
     return <div className="p-3">
-        <h1 className="font-bold text-3xl">{t('timeline')}</h1>
+            <div className="grid-cols-1 grid md:grid-cols-[1fr_auto] ">
+                <h1 className="font-bold text-3xl">{t('timeline')}</h1>
+                <div>
+                    <div className="grid grid-cols-[1fr_auto] gap-2">
+                        <span className="grid">{t('onlyFavored')}</span>
+                        <div className="static">
+                            <Switcher checked={filter.onlyFavored} setChecked={()=>dispatch(setFilters({...filter as Filter, onlyFavored: !filter?.onlyFavored}))}/>
+                        </div>
+                    </div>
+                </div>
+            </div>
         <div className="">{
             Object.keys(mappedEpisodes).map((e)=> {
                 // @ts-ignore

--- a/ui/src/pages/Timeline.tsx
+++ b/ui/src/pages/Timeline.tsx
@@ -1,14 +1,16 @@
 import {useTranslation} from "react-i18next";
 import {useAppDispatch, useAppSelector} from "../store/hooks";
-import {useEffect, useMemo} from "react";
+import {useEffect, useMemo, useState} from "react";
 import axios, {AxiosResponse} from "axios";
 import {apiURL, formatTime, getFiltersDefault} from "../utils/Utilities";
-import {setFilters, setTimeLineEpisodes} from "../store/CommonSlice";
+import {addTimelineEpisodes, setFilters, setTimeLineEpisodes} from "../store/CommonSlice";
 import {PodcastEpisodeTimeLine} from "../components/PodcastEpisodeTimeLine";
 import {TimeLineModel} from "../models/TimeLineModel";
 import {Switcher} from "../components/Switcher";
 import {Filter} from "../models/Filter";
 import {Loading} from "../components/Loading";
+import {Waypoint} from "react-waypoint";
+import {store} from "../store/store";
 
 const convertToTimeLineEpisodes = (podcastEpisodes: TimeLineModel[]) => {
     return podcastEpisodes.reduce((groups, game) => {
@@ -24,56 +26,84 @@ const convertToTimeLineEpisodes = (podcastEpisodes: TimeLineModel[]) => {
     }, {});
 }
 
-export const Timeline = ()=>{
+export const Timeline = () => {
     const {t} = useTranslation()
     const dispatch = useAppDispatch()
-    const timeLineEpisodes = useAppSelector(state=>state.common.timeLineEpisodes)
-    const mappedEpisodes = useMemo(()=>convertToTimeLineEpisodes(timeLineEpisodes),[timeLineEpisodes])
-    const filter = useAppSelector(state=>state.common.filters)
+    const timeLineEpisodes = useAppSelector(state => state.common.timeLineEpisodes)
+    const mappedEpisodes = useMemo(() => convertToTimeLineEpisodes(timeLineEpisodes), [timeLineEpisodes])
+    const filter = useAppSelector(state => state.common.filters)
+    let seenEpisodes = 0
+    let totalEpisodesGrouped = 0
 
-    useEffect(()=>{
-        !filter&&axios.get(apiURL+"/podcasts/filter")
-            .then((filterAxiosResponse:AxiosResponse<Filter>)=>{
-                    filterAxiosResponse.data == null && dispatch(setFilters(getFiltersDefault()))
+    useEffect(() => {
+        !filter && axios.get(apiURL + "/podcasts/filter")
+            .then((filterAxiosResponse: AxiosResponse<Filter>) => {
+                filterAxiosResponse.data == null && dispatch(setFilters(getFiltersDefault()))
 
                 dispatch(setFilters(filterAxiosResponse.data))
             })
     }, [])
 
-    useEffect(()=>{
-        if (filter){
-            axios.get(apiURL+"/podcasts/timeline",{
-                params:{
+    useEffect(() => {
+        if (filter) {
+            axios.get(apiURL + "/podcasts/timeline", {
+                params: {
                     favoredOnly: filter.onlyFavored
                 }
             })
-                .then((c:AxiosResponse<TimeLineModel[]>)=>{
+                .then((c: AxiosResponse<TimeLineModel[]>) => {
                     dispatch(setTimeLineEpisodes(c.data))
                 })
         }
-    },[filter])
+    }, [filter])
 
-    if (filter == null){
+    if (filter == null) {
         return <Loading/>
     }
 
     return <div className="p-3">
-            <div className="grid-cols-1 grid md:grid-cols-[1fr_auto] ">
-                <h1 className="font-bold text-3xl">{t('timeline')}</h1>
-                <div>
-                    <div className="grid grid-cols-[1fr_auto] gap-2">
-                        <span className="grid">{t('onlyFavored')}</span>
-                        <div className="static">
-                            <Switcher checked={filter.onlyFavored} setChecked={()=>dispatch(setFilters({...filter as Filter, onlyFavored: !filter?.onlyFavored}))}/>
-                        </div>
+        <div className="grid-cols-1 grid md:grid-cols-[1fr_auto] ">
+            <h1 className="font-bold text-3xl">{t('timeline')}</h1>
+            <div>
+                <div className="grid grid-cols-[1fr_auto] gap-2">
+                    <span className="grid">{t('onlyFavored')}</span>
+                    <div className="static">
+                        <Switcher checked={filter.onlyFavored} setChecked={() => dispatch(setFilters({
+                            ...filter as Filter,
+                            onlyFavored: !filter?.onlyFavored
+                        }))}/>
                     </div>
                 </div>
             </div>
+        </div>
         <div className="">{
-            Object.keys(mappedEpisodes).map((e)=> {
-                // @ts-ignore
-                let episodesOnDate = mappedEpisodes[e] as TimeLineModel[]
-                return <div key={e} className="bg-gray-800 mb-5 p-3 rounded"><h2 className="text-xl text-white">{formatTime(e)}</h2><div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4 gap-4">{episodesOnDate.map((v)=><PodcastEpisodeTimeLine podcastEpisode={v} key={v.podcast_episode.episode_id+"Parent"}/>)}</div></div>
+            Object.keys(mappedEpisodes).map((e, index) => {
+                    // @ts-ignore
+                    let episodesOnDate = mappedEpisodes[e] as TimeLineModel[]
+                    seenEpisodes = seenEpisodes + episodesOnDate.length
+                    totalEpisodesGrouped = Object.keys(mappedEpisodes).length
+                    return <div key={e} className="bg-gray-800 mb-5 p-3 rounded"><h2
+                        className="text-xl text-white">{formatTime(e)}</h2>
+                        {
+                            seenEpisodes == totalEpisodesGrouped&&<Waypoint key={index+"waypoint"} onEnter={()=>{
+                                axios.get(apiURL+"/podcasts/timeline", {
+                                    params:{
+                                        lastTimestamp: episodesOnDate[episodesOnDate.length-1].podcast_episode.date_of_recording,
+                                        favoredOnly: store.getState().common.filters?.onlyFavored
+                                    }
+                                })
+                                    .then((response:AxiosResponse<TimeLineModel[]>)=>{
+                                        dispatch(addTimelineEpisodes(response.data))
+                                    })
+                            }
+                            }/>
+                        }
+                        <div
+                            className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4 gap-4">{episodesOnDate.map((v) =>
+                            <PodcastEpisodeTimeLine podcastEpisode={v} key={v.podcast_episode.episode_id+index + "Parent"}
+                                                    index={index} timelineLength={timeLineEpisodes.length}/>)}
+                        </div>
+                    </div>
                 }
             )
 

--- a/ui/src/store/CommonSlice.ts
+++ b/ui/src/store/CommonSlice.ts
@@ -7,6 +7,7 @@ import {User} from "../models/User";
 import {ConfirmModalProps} from "../components/ConfirmModal";
 import {Invite} from "../pages/InviteAdministrationUserPage";
 import {TimeLineModel} from "../models/TimeLineModel";
+import {Filter} from "../models/Filter";
 
 export type Podcast = {
     directory: string,
@@ -57,7 +58,8 @@ interface CommonProps {
     users: User[],
     addInviteModalOpen: boolean,
     invites: Invite[],
-    timeLineEpisodes: TimeLineModel[]
+    timeLineEpisodes: TimeLineModel[],
+    filters: Filter|undefined
 }
 
 // Define the initial state using that type
@@ -78,7 +80,8 @@ const initialState: CommonProps = {
     users: [],
     addInviteModalOpen: false,
     invites: [],
-    timeLineEpisodes:[]
+    timeLineEpisodes:[],
+    filters: undefined
 }
 
 export const commonSlice = createSlice({
@@ -164,10 +167,13 @@ export const commonSlice = createSlice({
         },
         addPodcastEpisodes: (state, action:PayloadAction<PodcastEpisode[]>) => {
             state.selectedEpisodes = [...state.selectedEpisodes, ...action.payload]
+        },
+        setFilters: (state, action:PayloadAction<Filter>) => {
+            state.filters = action.payload
         }
 }})
 
-export const {setSideBarCollapsed, addPodcastEpisodes,setTimeLineEpisodes,setInvites,setAddInviteModalOpen, setUsers, setSelectedUser, setConfirmModalData,setLoginData, addPodcast, setCurrentDetailedPodcastId, setConfigModel, setPodcasts,setSelectedEpisodes, setSearchedPodcasts,updateLikePodcast, setInfoModalDownloaded,
+export const {setSideBarCollapsed, setFilters, addPodcastEpisodes,setTimeLineEpisodes,setInvites,setAddInviteModalOpen, setUsers, setSelectedUser, setConfirmModalData,setLoginData, addPodcast, setCurrentDetailedPodcastId, setConfigModel, setPodcasts,setSelectedEpisodes, setSearchedPodcasts,updateLikePodcast, setInfoModalDownloaded,
     setNotifications, removeNotification, setInfoModalPodcast, setInfoModalPodcastOpen, setDetailedAudioPlayerOpen} = commonSlice.actions
 
 export default commonSlice.reducer

--- a/ui/src/store/CommonSlice.ts
+++ b/ui/src/store/CommonSlice.ts
@@ -165,6 +165,9 @@ export const commonSlice = createSlice({
         setTimeLineEpisodes: (state, action:PayloadAction<TimeLineModel[]>) => {
             state.timeLineEpisodes = action.payload
         },
+        addTimelineEpisodes: (state, action:PayloadAction<TimeLineModel[]>) => {
+            state.timeLineEpisodes = [...state.timeLineEpisodes, ...action.payload]
+        },
         addPodcastEpisodes: (state, action:PayloadAction<PodcastEpisode[]>) => {
             state.selectedEpisodes = [...state.selectedEpisodes, ...action.payload]
         },
@@ -173,7 +176,7 @@ export const commonSlice = createSlice({
         }
 }})
 
-export const {setSideBarCollapsed, setFilters, addPodcastEpisodes,setTimeLineEpisodes,setInvites,setAddInviteModalOpen, setUsers, setSelectedUser, setConfirmModalData,setLoginData, addPodcast, setCurrentDetailedPodcastId, setConfigModel, setPodcasts,setSelectedEpisodes, setSearchedPodcasts,updateLikePodcast, setInfoModalDownloaded,
+export const {setSideBarCollapsed, addTimelineEpisodes,setFilters, addPodcastEpisodes,setTimeLineEpisodes,setInvites,setAddInviteModalOpen, setUsers, setSelectedUser, setConfirmModalData,setLoginData, addPodcast, setCurrentDetailedPodcastId, setConfigModel, setPodcasts,setSelectedEpisodes, setSearchedPodcasts,updateLikePodcast, setInfoModalDownloaded,
     setNotifications, removeNotification, setInfoModalPodcast, setInfoModalPodcastOpen, setDetailedAudioPlayerOpen} = commonSlice.actions
 
 export default commonSlice.reducer

--- a/ui/src/utils/Utilities.tsx
+++ b/ui/src/utils/Utilities.tsx
@@ -1,17 +1,18 @@
 import axios from "axios";
 import TimeAgo from 'javascript-time-ago'
-import sanitizeHtml,{IOptions} from 'sanitize-html'
+import sanitizeHtml, {IOptions} from 'sanitize-html'
 import en from 'javascript-time-ago/locale/en'
 import de from 'javascript-time-ago/locale/de'
 import fr from 'javascript-time-ago/locale/fr'
 import i18n from "i18next";
 import {PodcastEpisode} from "../store/CommonSlice";
 import {PodcastWatchedModel} from "../models/PodcastWatchedModel";
+import {Filter} from "../models/Filter";
 
 const defaultOptions: IOptions = {
-    allowedTags: [ 'b', 'i', 'em', 'strong', 'a' ],
+    allowedTags: ['b', 'i', 'em', 'strong', 'a'],
     allowedAttributes: {
-        'a': [ 'href' ]
+        'a': ['href']
     },
     allowedIframeHostnames: ['www.youtube.com']
 };
@@ -38,29 +39,28 @@ export const isLocalhost = Boolean(
 
 export let apiURL: string
 export let uiURL: string
-    if (window.location.pathname.startsWith("/ui")) {
-        apiURL = window.location.protocol + "//" + window.location.hostname + ":" + window.location.port+"/api/v1"
-    }
-    else {
-        //match everything before /ui
-        const regex = /\/([^/]+)\/ui\//
-        apiURL = window.location.protocol + "//" + window.location.hostname + ":" + window.location.port + "/" + regex.exec(window.location.href)![1] + "/api/v1"
-    }
-    uiURL =window.location.protocol+"//"+window.location.hostname+":"+window.location.port+"/ui"
+if (window.location.pathname.startsWith("/ui")) {
+    apiURL = window.location.protocol + "//" + window.location.hostname + ":" + window.location.port + "/api/v1"
+} else {
+    //match everything before /ui
+    const regex = /\/([^/]+)\/ui\//
+    apiURL = window.location.protocol + "//" + window.location.hostname + ":" + window.location.port + "/" + regex.exec(window.location.href)![1] + "/api/v1"
+}
+uiURL = window.location.protocol + "//" + window.location.hostname + ":" + window.location.port + "/ui"
 
 const wsEndpoint = "ws"
 
 export const configWSUrl = (url: string) => {
-    if(url.startsWith("http")){
-        return url.replace("http","ws")+wsEndpoint
+    if (url.startsWith("http")) {
+        return url.replace("http", "ws") + wsEndpoint
     }
-    return url.replace("https","wss")+wsEndpoint
+    return url.replace("https", "wss") + wsEndpoint
 }
-export  const logCurrentPlaybackTime = (episodeId: string,timeInSeconds: number)=> {
-        axios.post(apiURL+"/podcast/episode", {
-            podcastEpisodeId: episodeId,
-            time: Number(timeInSeconds.toFixed(0))
-        })
+export const logCurrentPlaybackTime = (episodeId: string, timeInSeconds: number) => {
+    axios.post(apiURL + "/podcast/episode", {
+        podcastEpisodeId: episodeId,
+        time: Number(timeInSeconds.toFixed(0))
+    })
 }
 
 export const formatTime = (isoDate: string) => {
@@ -69,7 +69,7 @@ export const formatTime = (isoDate: string) => {
 
 export const removeHTML = (html: string) => {
     return {
-        __html:sanitizeHtml(html, defaultOptions)
+        __html: sanitizeHtml(html, defaultOptions)
     }
 }
 
@@ -84,34 +84,47 @@ export const isJsonString = (str: string) => {
 }
 
 
-export const capitalizeFirstLetter = (string: string|undefined)=> {
-    if(string === undefined) return ""
+export const capitalizeFirstLetter = (string: string | undefined) => {
+    if (string === undefined) return ""
     return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
-    export const preparePath = (path: string|undefined) => {
-        if(path === undefined) return ""
+export const preparePath = (path: string | undefined) => {
+    if (path === undefined) return ""
 
-        return window.location.href.substring(0,window.location.href.indexOf('ui/'))+path.replaceAll(' ','%20').replaceAll('#','%23')
+    return window.location.href.substring(0, window.location.href.indexOf('ui/')) + path.replaceAll(' ', '%20').replaceAll('#', '%23')
+}
+
+export const preparePodcastEpisode = (episode: PodcastEpisode, response: PodcastWatchedModel) => {
+    return {
+        ...episode,
+        local_url: preparePath(episode.local_url),
+        local_image_url: preparePath(episode.local_image_url),
+        time: response.watchedTime
     }
+}
 
-    export const preparePodcastEpisode = (episode: PodcastEpisode, response: PodcastWatchedModel) => {
-    return {...episode,
-            local_url: preparePath(episode.local_url),
-            local_image_url: preparePath(episode.local_image_url),
-            time: response.watchedTime
+
+export const prepareOnlinePodcastEpisode = (episode: PodcastEpisode, response: PodcastWatchedModel) => {
+    const strippedUrl = episode.url.replace(/\?.*$/, "");
+    const strippedImage = episode.image_url.replace(/\?.*$/, "");
+    const test = window.location.href.substring(0, window.location.href.indexOf('ui/')) + 'proxy/podcast?url=' + strippedUrl
+
+    return {
+        ...episode,
+        local_url: test,
+        local_image_url: strippedImage,
+        time: response.watchedTime
     }
-    }
+}
 
+export const getFiltersDefault = () => {
+    return {
+        ascending: true,
+        filter: "PUBLISHEDDATE",
+        onlyFavored: false,
+        title: '',
+        username: ''
+    } satisfies Filter
+}
 
-    export const prepareOnlinePodcastEpisode = (episode: PodcastEpisode, response: PodcastWatchedModel)=>{
-        const strippedUrl = episode.url.replace(/\?.*$/, "");
-        const strippedImage = episode.image_url.replace(/\?.*$/, "");
-        const test =  window.location.href.substring(0,window.location.href.indexOf('ui/'))+ 'proxy/podcast?url='+strippedUrl
-
-        return {...episode,
-            local_url: test,
-            local_image_url: strippedImage,
-            time: response.watchedTime
-        }
-    }

